controllers/istiobase/istiobase_controller.go

```diff
@@ -0,0 +1,244 @@
+	"github.com/istio-ecosystem/sail-operator/pkg/enqueuelogger"
+	"github.com/istio-ecosystem/sail-operator/pkg/helm"
+	"github.com/istio-ecosystem/sail-operator/pkg/istioversion"
+	predicate2 "github.com/istio-ecosystem/sail-operator/pkg/predicate"
+	admissionv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	baseReleaseName = "istio-base"
+	baseChartName   = "base"
+
+	webhookName = "istiod-default-validator"
+)
+
+// Reconciler reconciles an IstioCNI object
+type Reconciler struct {
+	client.Client
+	Config       config.ReconcilerConfig
+	Scheme       *runtime.Scheme
+	ChartManager *helm.ChartManager
+}
+
+func NewReconciler(cfg config.ReconcilerConfig, client client.Client, scheme *runtime.Scheme, chartManager *helm.ChartManager) *Reconciler {
+	return &Reconciler{
+		Config:       cfg,
+		Client:       client,
+		Scheme:       scheme,
+		ChartManager: chartManager,
+	}
+}
+
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=validatingadmissionpolicies,verbs="*"
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=validatingadmissionpolicybindings,verbs="*"
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.1/pkg/reconcile
+func (r *Reconciler) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx)
+
+	reconcileErr := r.doReconcile(ctx)
+	log.Info("Reconciliation done.")
+
+	return ctrl.Result{}, reconcileErr
+}
+
+func (r *Reconciler) doReconcile(ctx context.Context) error {
+	log := logf.FromContext(ctx)
+
+	rev, err := r.getDefaultRevision(ctx)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if err := r.uninstallHelmChart(ctx); err != nil {
+				return fmt.Errorf("failed to uninstall base chart: %w", err)
+			}
+			return nil
+		}
+		return fmt.Errorf("failed to reconcile: %w", err)
+	}
+	log.Info("Installing Helm chart")
+	return r.installHelmChart(ctx, rev)
+}
+
+func (r *Reconciler) getDefaultRevision(ctx context.Context) (*v1.IstioRevision, error) {
+	// 1. get the IstioRevision referenced in the default IstioRevisionTag
+	tag := v1.IstioRevisionTag{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: "default"}, &tag); err != nil {
```

Comment:
This can only ever be `default`? In the helm installation of istio, it seems you can set `defaultRevision` to be another revision name. From: https://istio.io/latest/docs/setup/install/helm/#installation-steps

> When performing a revisioned installation, the base chart requires the --set defaultRevision=<revision> value to be set for resource validation to function. Below we install the default revision, so --set defaultRevision=default is configured.

Reply:
Yeah, technically you can, at least upstream - we need to deprecate that field in our API. If we're going to support it, individual `Istio` resources are probably not the place to configure it

Reply:
Already deprecated: https://github.com/istio-ecosystem/sail-operator/pull/760